src/webextension/list/manage/components/edit-item-details.js

```diff
@@ -43,6 +43,14 @@ export default class EditItemDetails extends React.Component {
     this.state = {...props.fields};
   }
 
+  componentWillReceiveProps(nextProps) {
+    for (let i in this.props.fields) {
+      if (this.props.fields[i] !== nextProps.fields[i]) {
+        this.setState({[i]: nextProps.fields[i]});
+      }
+    }
```

Comment:
There's a problem with this that I'm not sure how to fix. The idea here is that when the component gets new props, it'll update any field whose *initial* value has been change (this is to prevent bogus prop updates from resetting form state). However, if you really do want to reset a field even when it's the same thing (e.g. two entries with the same email address), it won't reset.

I'll try to see if there's a cleaner way to reset the form, but if that doesn't work out, maybe I'll just drop the first commit here.

Reply:
I was going to ask about that in my review.

Reply:
There should be a comment here explaining why this is being done, if this code is retained.